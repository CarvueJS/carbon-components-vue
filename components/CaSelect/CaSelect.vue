<template>
  <div class="bx--select">
    <select id="select-id" class="bx--select-input">
      <slot></slot>
    </select>
    <svg class="bx--select__arrow" width="10" height="5" viewBox="0 0 10 5" fill-rule="evenodd">
      <path d="M10 0L5 5 0 0z"></path>
    </svg>
  </div>
</template>

<script>
  export default {
    name: 'ca-select',
  };
</script>
