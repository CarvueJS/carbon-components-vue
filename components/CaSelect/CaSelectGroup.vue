<template>
  <optgroup class="bx--select-optgroup" :label="label">
    <slot></slot>
  </optgroup>
</template>

<script>
  export default {
    name: 'ca-select-group',
    props: {
      label: {
        type: String,
        default: '',
      },
    },
  };
</script>

<style></style>
