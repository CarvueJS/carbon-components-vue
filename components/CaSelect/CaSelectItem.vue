<template>
  <option class="bx--select-option" :value="value">
    <slot></slot>
  </option>
</template>

<script>
  export default {
    name: 'ca-select-item',
    props: {
      value: {
        type: String,
        default: '',
      },
    },
  };
</script>

<style></style>
