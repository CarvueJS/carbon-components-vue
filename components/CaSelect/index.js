import CaSelect from './CaSelect.vue';
import CaSelectItem from './CaSelectItem.vue';
import CaSelectGroup from './CaSelectGroup.vue';

CaSelect.install = Vue => {
  Vue.component(CaSelect.name, CaSelect);
  Vue.component(CaSelectItem.name, CaSelectItem);
  Vue.component(CaSelectGroup.name, CaSelectGroup);
}

export default CaSelect
