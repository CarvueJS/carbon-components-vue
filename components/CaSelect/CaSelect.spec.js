import { shallow } from 'vue-test-utils';
import CaSelect from './CaSelect.vue';

describe('CaSelect', () => {
  test('is a Vue instance', () => {
    const wrapper = shallow(CaSelect);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('Mathes snapshot', () => {
    const wrapper = shallow(CaSelect);
    expect(wrapper.html()).toMatchSnapshot()
  });
})
