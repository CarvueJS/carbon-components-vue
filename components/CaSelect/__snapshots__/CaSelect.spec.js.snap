// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaSelect Mathes snapshot 1`] = `
<div class="bx--select">
  <select id="select-id" class="bx--select-input"></select>
  <svg width="10" height="5" viewBox="0 0 10 5" fill-rule="evenodd" class="bx--select__arrow">
    <path d="M10 0L5 5 0 0z"></path>
  </svg>
</div>
`;
